packages/react/src/components/Checkbox/CheckboxBase.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+
+interface CheckboxBaseProps {
+  checked?: boolean
+  onChange?: React.ChangeEventHandler
+  onClick?: React.MouseEventHandler
+  slots?: {
+    root?: any
+    input?: any
+  }
+  slotProps?: {
+    root?: any
+    input?: any
+  }
+  classes?: any
+}
+
+const Testhelper = (prop?: Function, propName?: string) => {
+  if (!prop) {
+    return {}
+  }
+  return {
+    [propName]: prop,
+  }
+}
+
+const CheckboxBase: React.FunctionComponent<CheckboxBaseProps> = props => {
+  const { checked, onChange, onClick, classes = {}, slots = {}, slotProps = {}, ...rest } = props
+  const { root: RootSlot = 'div', input: InputSlot = 'input' } = slots
+  const { root: rootClass, input: inputClass } = classes
+  const { root: rootProps, input: inputProps } = slotProps
+
+  const [isChecked, setIsChecked] = React.useState(!!checked)
+  const realIsChecked = checked === undefined ? isChecked : !!checked
+
+  const onChangeHandler = React.useCallback(
+    (ev: any) => {
+      if (onClick) {
+        onClick(ev)
+      }
+      if (!ev.defaultPrevented) {
+        if (onChange) {
+          onChange(ev)
+        }
+      }
+      if (!ev.defaultPrevented) {
+        setIsChecked(!realIsChecked)
+      }
+    },
+    [setIsChecked, realIsChecked, onChange],
```

Comment:
Do we need to add `setIsChecked` here? From my understanding, since the function will never change its value, it is a variable that will never trigger a different callback.